flake.nix

```diff
@@ -186,26 +217,36 @@
               export PLAYWRIGHT_SKIP_VALIDATE_HOST_REQUIREMENTS=true
             '';
 
-            LOCALE_ARCHIVE = with pkgs; lib.optionalDrvAttr stdenv.isLinux "${glibcLocales}/lib/locale/locale-archive";
+            LOCALE_ARCHIVE =
+              with pkgs;
+              lib.optionalDrvAttr stdenv.isLinux "${glibcLocales}/lib/locale/locale-archive";
           };
         };
 
-        packages = {
-          proto_gen_go = proto_gen_go_1_30;
-          all = pkgs.buildEnv {
-            name = "all-packages";
-            paths = devShellPackages;
-          };
-          site = buildSite;
-
-          # Copying `OS_ARCHES` from the Makefile.
-          linux_amd64 = buildFat "linux_amd64";
-          linux_arm64 = buildFat "linux_arm64";
-          darwin_amd64 = buildFat "darwin_amd64";
-          darwin_arm64 = buildFat "darwin_arm64";
-          windows_amd64 = buildFat "windows_amd64.exe";
-          windows_arm64 = buildFat "windows_arm64.exe";
-        };
+        packages =
+          {
+            default = packages.${system};
+
+            site = buildSite;
+
+            # Copying `OS_ARCHES` from the Makefile.
+            x86_64-linux = buildFat "linux_amd64";
+            aarch64-linux = buildFat "linux_arm64";
+            x86_64-darwin = buildFat "darwin_amd64";
+            aarch64-darwin = buildFat "darwin_arm64";
+            x86_64-windows = buildFat "windows_amd64.exe";
+            aarch64-windows = buildFat "windows_arm64.exe";
+          }
+          // (pkgs.lib.optionalAttrs pkgs.stdenv.isLinux {
+            dev_image = pkgs.dockerTools.buildNixShellImage {
+              name = "codercom/oss-dogfood-nix";
+              tag = "latest-${system}";
+
+              drv = devShells.default.overrideAttrs (oldAttrs: {
+                buildInputs = oldAttrs.buildInputs ++ [ pkgs.nix ];
+              });
```

Comment:
Is this only expected to work on Linux systems?

Reply:
Yes, building an OCI image is only supported on Linux systems.

Reply:
Personally/locally, I've spun up a VM with nixOS and configured it as a remote builder, but any other Linux OS will do.

It's the same thing that Docker Desktop/Colima/OrbStack/... do for one under the hood but without the hood and magic abstractions part.